source/user/mcscf.rst

```diff
@@ -1,57 +1,329 @@
-number of alpha electrons is equal to the number of beta electrons,
-since the ``mol`` object is initialized with ``spin=0``.  The spin
-multiplicity of the CASSCF/CASCI solver can be changed by the fourth
-argument::
 
-  >>> mc = mcscf.CASSCF(m, 4, (4,2))
-  >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-  E(CASSCF) = -149.609461
-  >>> print('S^2 = %.7f, 2S+1 = %.7f' % mcscf.spin_square(mc))
-  S^2 = 2.0000000, 2S+1 = 3.0000000
+CASSCF
+""""""
+.. code-block:: python
 
-The two integers in the tuple represent the number of alpha and beta electrons.
-Although it is a triplet state, the solution might not be correct since the
-CASSCF is based on the incorrect singlet HF ground state.  Starting from the
-ROHF ground state, we have::
+  import pyscf
+  mol = pyscf.M(
+      atom = 'O 0 0 0; O 0 0 1.2',
+      basis = 'ccpvdz',
+      spin = 2)
+  myhf = mol.RHF().run()
+  ncas, nelecas = (6,(5,3))
+  mycas = myhf.CASSCF(ncas, nelecas).run()
 
-  >>> mc = mcscf.CASSCF(rhf3, 4, 6)
-  >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-  E(CASSCF) = -149.646746
+Like many other modules in PySCF, :mod:`mcscf` works with density-fitting (:numref:`user_df`) and x2c (:numref:`user_x2c`).
+It can also be used in its unrestricted formulation, please see :source:`examples/mcscf/60-uhf_based_ucasscf.py` for an example.
+An important feature of :mod:`mcscf` is that it can interface with external CI solver such as DMRG, FCIQMC, or selected CI methods, see the external projects for more details.
+You can even use MP2 or CC methods as solvers!
+Check out :source:`/examples/cc/42-as_casci_fcisolver.py` for a working example.
+
+
+Picking an Active Space
+-----------------------
+In general, selecting an active space can be cumbersome and PySCF offers severals ways to facilitate this process.
+There are several strategies and they all contain two main components:
+
+* Specify the number of electrons and orbitals in the active space.
+* (Optional) Specify the which orbitals are "active" 
 
-The energy is lower than the RHF initial guess.
-.. We can also use the UHF ground
-.. state to start a CASSCF calculation::
-.. 
-..   >>> mc = mcscf.CASSCF(uhf3, 4, 6)
-..   >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-..   E(CASSCF) = -149.661324
-..   >>> print('S^2 = %.7f, 2S+1 = %.7f' % mcscf.spin_square(mc))
-..   S^2 = 3.9713105, 2S+1 = 4.1091656
 .. 
-.. Woo, the total energy is even lower.  But the spin is contaminated.
+  warning::
+  The total set orbitals (core, active, and virtual) used in active space methods can be specified or selected in a variety of ways, giving users substantial flexibility for CAS-type calculations.
+  But users should note, "with great power comes great responsibility."
+  Active space calculations are notoriously difficult and just because a calculation completes without error does not guarantee that the results will be chemically/physically meaningful so we urge users to select their active space orbitals with thought and care.
```

Comment:
```suggestion
  Active space calculations are notoriously difficult and just because a calculation completes without error does not guarantee that the results will be chemically/physically meaningful.
  We urge users to select their active space orbitals with thought and care.
  You should always try out several choices for the active space!
```